Slightly random tweaks & bug fixes

TheKodeToad opened the pull request:
Sorry if this is annoying to review, I kind of got carried away :p

TheKodeToad committed:
A few tweaks, with inspiration from Zeke :3

Signed-off-by: TheKodeToad <TheKodeToad@proton.me>

TheKodeToad committed:
Fix flat white launcher icon

Signed-off-by: TheKodeToad <TheKodeToad@proton.me>

TheKodeToad committed:
Fix Open Global Settings, why not

Signed-off-by: TheKodeToad <TheKodeToad@proton.me>

TheKodeToad committed:
Revert "feat(Mods): hide 'Provider' column when no mods have providers"

With Ryex's change, this causes issues. Apparently you need to sign off reverts! That's just weird...

Signed-off-by: TheKodeToad <TheKodeToad@proton.me>

leia-uwu commented on launcher/resources/flat_white/scalable/launcher.svg:
does the launcher icon needs to be duplicated on all icon themes? iirc the "legacy" / multimc icon theme is a fallback for all themes but not sure

DioEgizio replied:
Should be

TheKodeToad replied:
I was going to delete the icon, then I realised the other themes duplicated it.

leia-uwu replied:
imo we should remove the icon for the other icon themes, theres no reason to duplicate it on all themes :p

TheKodeToad committed:
Replace accidental usages of QAbstractButton::pressed

This signal is not usually what you want, and creates an inconsistent experience.

Signed-off-by: TheKodeToad <TheKodeToad@proton.me>

TheKodeToad committed:
Fix big mistake :iea:

Signed-off-by: TheKodeToad <TheKodeToad@proton.me>

TheKodeToad committed:
Fix markdown header

Signed-off-by: TheKodeToad <TheKodeToad@proton.me>

TheKodeToad committed:
Deduplicate launcher icon

Signed-off-by: TheKodeToad <TheKodeToad@proton.me>

Scrumplex committed:
fix: install logo to multimc theme in genicons.sh

Signed-off-by: Sefa Eyeoglu <contact@scrumplex.net>

Ryex commented on launcher/ui/widgets/ModListView.cpp:
```diff
@@ -65,19 +62,6 @@ void ModListView::setModel ( QAbstractItemModel* model )
         for(int i = 1; i < head->count(); i++)
             head->setSectionResizeMode(i, QHeaderView::ResizeToContents);
     }
-
-    auto real_model = model;
-    if (auto proxy_model = dynamic_cast<QSortFilterProxyModel*>(model); proxy_model)
-        real_model = proxy_model->sourceModel();
-
-    if (auto mod_model = dynamic_cast<ModFolderModel*>(real_model); mod_model) {
-        connect(mod_model, &ModFolderModel::updateFinished, this, [this, mod_model]{
-            auto mods = mod_model->allMods();
-            // Hide the 'Provider' column if no mod has a defined provider!
-            setColumnHidden(ModFolderModel::Columns::ProviderColumn,
-                    std::none_of(mods.constBegin(), mods.constEnd(), [](auto const mod){ return mod->provider().has_value(); }));
-        });
-    }
```
whoa! why are we removing this?

TheKodeToad replied:
Because it means the user's choice whether the provider column is hidden/shown is ignored

Ryex replied:
ah, right. the image column pr was merged so it's a proper setting now.

Ryex commented on program_info/genicons.sh:
```diff
@@ -67,7 +67,4 @@ else
-do
-    cp -v org.prismlauncher.PrismLauncher.svg "$dir/launcher.svg"
-done
+cp -v org.prismlauncher.PrismLauncher.svg "../launcher/resources/multimc/scalable/launcher.svg"
```
and on windows where bash/sh can't run?

Scrumplex replied:
/shrug

this works best on macOS anyway, as we need some macOS tools

TheKodeToad replied:
this pr specifically hasn't created any issues, so resolving this

Ryex commented on launcher/ui/pages/instance/VersionPage.ui:
```diff
@@ -204,14 +202,6 @@
     <string>Install the LiteLoader package.</string>
    </property>
   </action>
-  <action name="actionInstall_mods">
-   <property name="text">
-    <string>Install mods</string>
-   </property>
-   <property name="toolTip">
-    <string>Install normal mods.</string>
-   </property>
-  </action>
```
I take it these are no longer used?

TheKodeToad replied:
It's just a duplicate action; it takes you straight to the mods page
